NEW Add option to clone its default BOM when cloning a product

vmaury opened the pull request:
This option is available if 

- BOM module is enabled (elementary my dear Watson)
- product has a default BOM
- user has right to create BOM
![clone BOM](https://github.com/Dolibarr/dolibarr/assets/5389628/5655bc68-ab2f-42cf-9a04-274506d40f0e)

vmaury committed:
Add option to clone its default BOM when cloning a product

vmaury committed:
Improved label of cloned BOM ("BOM of cloned_product_ref")

eldy commented:
What is the default value of the new checkbox ? Should be "not" checked by default.

vmaury committed:
- option "clone default BOM" not checked by default ( can be pre-checked with constant BOM_CLONE_DEFBOM)
- add another option to clone all BOMs linked to a product (not checked by default)

vmaury commented:
Ok. Now :
- option "clone default BOM" not checked by default ( can be pre-checked with constant BOM_CLONE_DEFBOM)
- and I add another option to clone all BOMs linked to a product
![clone BOM2](https://github.com/Dolibarr/dolibarr/assets/5389628/40ec9ec0-0897-4124-a0d8-e8ad2b86f56d)

vmaury committed:
- option "clone default BOM" not checked by default ( can be pre-checked with constant BOM_CLONE_DEFBOM)
- add another option to clone all BOMs linked to a product (not checked by default)

eldy committed:
Update mrp.lang

eldy committed:
Update mrp.lang
